synth/src/utils.rs

```diff
@@ -35,6 +38,53 @@ pub fn version() -> String {
     env!("CARGO_PKG_VERSION").to_string()
 }
 
+/// Notify the user if there is a new version of Synth
+/// Even though the error is not meant to be used, it
+/// makes the implementation simpler instead of returning ().
+pub fn notify_new_version() -> Result<()> {
+    let current_version = crate::utils::version();
+    let latest_version = latest_version()?;
+    notify_new_version_inner(&current_version, &latest_version)
+}
+
+fn latest_version() -> Result<String> {
+    let url = "https://api.github.com/repos/getsynth/synth/releases/latest";
+    let client = reqwest::blocking::Client::new();
+    let response = client
+        .get(url)
+        .header(USER_AGENT, "hyper/0.14")
+        .timeout(Duration::from_secs(2))
+        .send()?;
+
+    let release_info: Map<String, Value> = response.json()?;
+
+    // We're assuming here that the GH API doesn't make breaking changes
+    // otherwise these `get` and `as_str` operations are quite safe
+    let latest_version = release_info
+        .get("name")
+        .ok_or(anyhow!("Could not get the 'name' parameter"))?
+        .as_str()
+        .ok_or(anyhow!("was expecting name to be a string"))?;
+
+    // At this point it looks like 'vX.Y.Z'. Here we're removing the `v`
+    // Maybe we should use something that doesn't panic?
+    Ok(latest_version[1..].to_string())
+}
+
+pub fn notify_new_version_inner(current_version: &str, latest_version: &str) -> Result<()> {
+    // semver should be ok with lexicographical ordering, right?
+    if latest_version > current_version {
```

Comment:
Semver is not in general lexicographically ordered, e.g. `0.10.1` comes before `0.9.1`. I would suggest parsing the versions with the [semver](https://docs.rs/semver) crate and comparing the resulting `Version`s. This is guaranteed to work correctly.

Reply:
Good catch! Thank @llogiq .